Unity/Champloo/Assets/Scripts/Player/Movement States/Movement Specials/OnInfiniteDash.cs

```diff
@@ -4,25 +4,33 @@
 public class OnInfiniteDash : OnDash
 {
     private bool hitDashEarly;
+    private bool attackAllowed;
+
+    [SerializeField]
+    private float attackBufferWindow;
+    
     protected override void Start()
     {
         base.Start();
         hitDashEarly = false;
+        attackAllowed = false;
     }
 
-    public override bool AttackAllowed { get { return true; } }
+    public override bool AttackAllowed { get { return attackAllowed; } }
 
     public override MovementState DecideNextState(Vector3 velocity, Vector3 externalForces)
     {
         specialTimeLeft -= Time.deltaTime;
         //Buffer for attacks and movement specials done too early
+
+        if (specialTimeLeft < (specialTime * attackBufferWindow) && player.InputPlayer.GetButtonDown("Attack"))
```

Comment:
why? It needs to check for an attack input at some point and having a nested if statement felt unnecessary.